packages/local_auth/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -269,13 +271,19 @@ private void getAvailableBiometrics(final Result result) {
     if (activity == null || activity.isFinishing()) {
       return biometrics;
     }
+
+    if (!canAuthenticateWithBiometrics()) {
+      biometrics.add("undefined");
```

Comment:
You are returning "undefined" even on a device that has no hardware, which is wrong.

That means you are also missing critical test cases.